hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/CORSITest.groovy

```diff
@@ -0,0 +1,171 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.rest
+
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_ALLOW_CREDENTIALS
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_ALLOW_HEADERS
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_ALLOW_METHODS
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_ALLOW_ORIGIN
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_MAX_AGE
+import static org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_REQUEST_METHOD
+import static org.hawkular.jaxrs.filter.cors.Headers.DEFAULT_CORS_ACCESS_CONTROL_ALLOW_HEADERS
+import static org.hawkular.jaxrs.filter.cors.Headers.DEFAULT_CORS_ACCESS_CONTROL_ALLOW_METHODS
+import static org.hawkular.jaxrs.filter.cors.Headers.ORIGIN
+import static org.junit.Assert.assertEquals
+import static org.junit.Assume.assumeTrue
+
+import org.hawkular.alerts.api.model.event.Event
+
+import org.junit.Before
+import org.junit.Test
+import org.junit.Ignore
+
+class CORSITest extends AbstractITestBase {
+  static final String testOrigin = System.getProperty("hawkular.test.origin", "http://test.hawkular.org")
+  static final String testAccessControlAllowHeaders = DEFAULT_CORS_ACCESS_CONTROL_ALLOW_HEADERS + ',' +
+    System.getProperty("hawkular.test.access-control-allow-headers");
+
+  @Test
+  void testOptionsWithOrigin() {
+    def response = client.options(path: "ping",
+        headers: [
+            (ACCESS_CONTROL_REQUEST_METHOD): "POST",
+            //this should be ignored by the container and reply with pre-configured headers
+            (ACCESS_CONTROL_ALLOW_HEADERS): "test-header",
+            (ORIGIN): testOrigin,
+        ])
+
+    def responseHeaders = "==== Response Headers = Start  ====\n"
+    response.headers.each { responseHeaders += "${it.name} : ${it.value}\n" }
+    responseHeaders += "==== Response Headers = End ====\n"
+
+    //Expected a 200 because this is be a pre-flight call that should never reach the resource router
+    assertEquals(200, response.status)
+    assertEquals(null, response.getData())
+    assertEquals(responseHeaders, DEFAULT_CORS_ACCESS_CONTROL_ALLOW_METHODS, response.headers[ACCESS_CONTROL_ALLOW_METHODS].value)
+    assertEquals(responseHeaders, testAccessControlAllowHeaders, response.headers[ACCESS_CONTROL_ALLOW_HEADERS].value)
+    assertEquals(responseHeaders, testOrigin, response.headers[ACCESS_CONTROL_ALLOW_ORIGIN].value)
+    assertEquals(responseHeaders, "true", response.headers[ACCESS_CONTROL_ALLOW_CREDENTIALS].value)
+    assertEquals(responseHeaders, (72 * 60 * 60) + "", response.headers[ACCESS_CONTROL_MAX_AGE].value)
```

Comment:
Minor one.
I saw that in commons too. I suggest to not use a magic number and introduce a constant on:
org.hawkular.jaxrs.filter.cors.Headers.ACCESS_CONTROL_MAX_AGE_DEFAULT_VALUE

More easy to review and maintain in the future.